trie/zk_trie_impl.go

```diff
@@ -534,10 +536,10 @@ func (mt *ZkTrieImpl) GetNode(nodeHash *zkt.Hash) (*Node, error) {
 	mt.lock.RLock()
 	defer mt.lock.RUnlock()
 
-	return mt.getNode(nodeHash)
+	return mt.getNode(nodeHash, nil)
 }
 
-func (mt *ZkTrieImpl) getNode(nodeHash *zkt.Hash) (*Node, error) {
+func (mt *ZkTrieImpl) getNode(nodeHash *zkt.Hash, node *Node) (*Node, error) {
```

Comment:
I am afraid such an entry would be a little tricky for understanding. I would advice use two entries rather than one:

+ `getNodeTo` applies the arguments above but never return another `*Node`, so it is

```go
func (mt *ZkTrieImpl) getNodeTo(nodeHash *zkt.Hash, node *Node) error
```

And `getNode` keep the original signature, just wrap `getNodeTo` by allocating a new Node object and return it:

```go
func (mt *ZkTrieImpl) getNode(nodeHash *zkt.Hash, node *Node) (*Node, error) {
    node := new(Node)
    err := mt.getNodeTo(nodeHash, node)
    return node, err
}
```

Reply:
Well, this is just a private API so I don't think it is that complicated. But regardless, updated in https://github.com/scroll-tech/zktrie/pull/29/commits/b2368b153ed761e3b34ec65b67c25f931036e7a8